installer/bundle/bundle_skel_Linux.sh

```diff
@@ -702,29 +581,25 @@ case "$installMode" in
 
     I)
         echo "Installing cross-platform agent ..."
-        if [ "$PLATFORM" = "SunOS" ]; then
-            check_if_pkg_is_installed scx
-            if [ $? -eq 0 ]; then
-                echo "ERROR: SCX package is already installed"
-                cleanup_and_exit 2
-            fi
-        fi
 
         if [ $PROVIDER_ONLY -eq 0 ]; then
             check_if_pkg_is_installed omi
             if [ $? -eq 0 ]; then
-                pkg_upd $OMI_PKG omi
-                # It is acceptable that this fails due to the new omi being 
-                # the same version (or less) than the one currently installed.
-                OMI_EXIT_STATUS=0
+                pkg_upd_list $OMI_PKG omi
             else
-                pkg_add $OMI_PKG omi
-                OMI_EXIT_STATUS=$?
+                pkg_add_list $OMI_PKG omi
             fi
         fi
 
-        pkg_add $OM_PKG scx
-        SCX_EXIT_STATUS=$?
+        pkg_add_list $OM_PKG scx
+
+        pkg_upd ${UPD_PKG_QUEUE}
+        # It is acceptable that this fails due to the new omi being 
+        # the same version (or less) than the one currently installed.
+        OMI_EXIT_STATUS=0
```

Comment:
This is kind of whacky. I know that this is how you got it - my bad.

The intention here, before my most recent changes, is that if OMI was already installed, don't error out for that. But now, with the queue, setting OMI_EXIT_STATUS to zero is just wrong (as it'll not catch SCX install issues).

I think, strictly, this will work (amazingly enough), given that if nothing gets added to the list at 588, then 596 will be ignored since the list is empty. But still, whacky.

My suggestion is just to test if it's installed (at 587) and, if it is, do the same version check that the regular upgrade path does. Then, right after 588, call pkg_upd to actually do the update (since all other code below uses pkg_add). But here, since you've done the pre-check, actually do signal an error in OMI_EXIT_STATUS (since an error might have occurred).